dbt/include/databricks/macros/materializations/incremental/strategies.sql

```diff
@@ -97,11 +100,26 @@ select {{source_cols_csv}} from {{ source_relation }}
       {% do predicates.append('FALSE') %}
   {% endif %}
 
-  merge into {{ target }} as DBT_INTERNAL_DEST
-      using {{ source }} as DBT_INTERNAL_SOURCE
-      on {{ predicates | join(' and ') }}
-      when matched then update set {{ get_merge_update_set(update_columns, on_schema_change, source_columns) }}
-      when not matched then insert {{ get_merge_insert(on_schema_change, source_columns) }}
+    merge
+        {%- if merge_with_schema_evolution %}
+        with schema evolution
+        {%- endif %}
+    into
+        {{ target }} as DBT_INTERNAL_DEST
+    using
+        {{ source }} as DBT_INTERNAL_SOURCE
+    on
+        {{ predicates | join('\n    and ') }}
+    {%- if not skip_matched_step %}
+    when matched
+        then update set
```

Comment:
Sorry for weird formatting, I cleaned up my experiments with match conditions, they will reside here between these two lines.

Reply:
Formatting with Jinja will typically be weird, because you have to go out of your way to get spacing to not look terrible.  I wouldn't worry too much about it.